packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -11,24 +11,23 @@ import { Input, Props as InputProps } from '../Input/Input';
 
 import { getComboboxStyles } from './getComboboxStyles';
 
-export type Value = string | number;
 export type Option = {
   label: string;
-  value: Value;
+  value: string;
   description?: string;
 };
 
 interface ComboboxProps
   extends Omit<InputProps, 'prefix' | 'suffix' | 'value' | 'addonBefore' | 'addonAfter' | 'onChange'> {
   onChange: (val: Option | null) => void;
-  value: Value | null;
+  value: string | null;
```

Comment:
maybe using a generic here is better?

```ts
interface ComboboxProps<T = string> {
  value: T | null
}
```

Reply:
Possibly, but at the moment I see no reason to support other value types. `number` is probably the one I could agree on.

Reply:
yeah i was thinking number mostly. i can see a common case of an array of somethings where they can be efficiently identified by index, without having to coerce the index to string 300k times :)

Reply:
Hmm, I tend to agree. I will play around with it a bit more, to make sure that the typings don't end up being confusing.

Reply:
I think number is fine also.

Like I clarified back in my thread, I think the important part is that the value should remain a primitive to avoid any footguns around equality.